File: tests/fixtures/sample.php
<?php

namespace Demo\Core;

interface LoggerInterface
{
    public function log(string $message): void;
}

trait WithContext
{
    private function writeInternal(): void {}
}

class Logger implements LoggerInterface
{
    public const VERSION = '1.0';
    public static $shared;

    public function log(string $message): void {}
}

function create_logger(): Logger
{
    return new Logger();
}

File: tests/fixtures/sample.rb
module Logging
  VERSION = "1.0"
  current_level = VERSION

  class Logger
    def self.build(name)
      Logger.new(name)
    end

    def initialize(name)
      @name = name
    end

    def log(message)
      puts message
    end
  end
end
